src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,39 @@
+package duke.task;
+
+public class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    // duke.task.Todo: Handle exceptional cases
+    public static Task createTask(String[] commandArgs) throws IllegalArgumentException {
+        String description = "";
+        String by = "";
+
+        int toggle = 0;
+        for (int i = 0; i < commandArgs.length; i++) {
+            if (commandArgs[i].equals("/by")) {
+                toggle = 1;
+                continue;
+            }
+            // current token is part of description
+            if (toggle == 0) {
+                description += commandArgs[i] + " ";
+            }
+            // current token is part of by
+            else if (toggle == 1) {
+                by += commandArgs[i] + " ";
+            }
+        }
```

Comment:
Deadline extends Task that means Deadline is a Task. Personally feel that createTask should not be in here but do let me know your views!